Disable gRPC connectivity and gRPC service in neuro-san.

andreidenissov-cog opened the pull request:
This PR is a first step in disabling gRPC connectivity in neuro-san and making it
an http-only service.
This will simplify moving ahead with adding other http-based protocols like MCP 
(and maybe A2A and whatever else will become popular)
http Tornado server now becomes first-class service, running in a main app thread.

d1donlydfink commented on neuro_san/service/http/server/http_server.py:
```diff
@@ -138,8 +148,6 @@ def __call__(self, other_server: AgentServer):
         self.logger.info({}, "Http server stopped.")
-        if other_server is not None:
-            other_server.stop()
 
```
You might be able to remove the AgentServer interface as well.

d1donlydfink reviewed (changes requested):
Can't just yank this without yanking the calling references and/or docs.

Things that need to be done:

* You might be able to remove the AgentService interface altogether.
* What to do about AGENT_REQUEST_LIMIT? It is actually useful for a multi-pod affair.
* Need to update Dockerfile for env vars that are no longer applicable
* tests/fixtures/music_nerd_pro/combination_responses_with_history_grpc.hocon
    specifically tests grpc.  Need to at least change this to http and/or get w/ Vince
* test/assessor/assessor.py also references grpc as a connection type
* Need to update docs:
    * main README.md in multiple places
    * test_case_hocon_reference.md


Consider instead turning off the basis of the gRPC functionality with a Franken-Switch (tm) but leaving the code
and removing the documentation/client/test aspects first.  (Outside-in, instead of inside-out like this PR is).
That way we can gauge the pitch of the hew and cry for taking the functionality away.

andreidenissov-cog commented:
> Can't just yank this without yanking the calling references and/or docs.
> 
> Things that need to be done:
> 
> * You might be able to remove the AgentService interface altogether.
   Not right now. I am not touching anything outside main server loop (almost) so it'll be easy to go back if necessary.

> * What to do about AGENT_REQUEST_LIMIT? It is actually useful for a multi-pod affair.
    That will need some effort. Tornado itself does not support this parameter.
    Files github issue #480  

> * Need to update Dockerfile for env vars that are no longer applicable
    Done. Docker image is building and starting.

> * tests/fixtures/music_nerd_pro/combination_responses_with_history_grpc.hocon
>   specifically tests grpc.  Need to at least change this to http and/or get w/ Vince
   "grpc" and "http" variants are copies except for connectivity parameter.
    Removed "grpc" one.

> * test/assessor/assessor.py also references grpc as a connection type
    Removed "grpc" option.

> * Need to update docs:
>   
>   * main README.md in multiple places
>   * test_case_hocon_reference.md
>
    Updated docs removing references to gRPC option.
 
> Consider instead turning off the basis of the gRPC functionality with a Franken-Switch (tm) but leaving the code and removing the documentation/client/test aspects first. (Outside-in, instead of inside-out like this PR is). That way we can gauge the pitch of the hew and cry for taking the functionality away.

andreidenissov-cog commented on neuro_san/deploy/Dockerfile:
```diff
@@ -74,15 +74,6 @@ RUN \
-# be the first port exposed in the Dockerfile.
-EXPOSE 30011
-
 # This is the port the service will accept http requests on.
```
No 30011 port no more.

andreidenissov-cog commented on neuro_san/test/assessor/assessor.py:
```diff
@@ -86,18 +86,17 @@ def add_args(self, arg_parser: argparse.ArgumentParser):
         arg_parser.add_argument("--connection", default="direct", type=str,
-                                choices=["grpc", "direct", "http", "https"],
+                                choices=["direct", "http", "https"],
                                 help="""
```
No "grpc" option.

d1donlydfink commented on docs/test_case_hocon_reference.md:
```diff
@@ -395,7 +394,7 @@ For numbers, this is the expected inequality. For strings, this is a lexicograph
 [TestCase.assertLess](https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertLess)
 
-Similarly the "not_less" test looks for a value that is considered greater than or equal to what is given.
+Similarly, the "not_less" test looks for a value that is considered greater than or equal to what is given.
```
thanks for the grammar checks.

andreidenissov-cog commented on neuro_san/session/async_grpc_service_agent_session.py:
```diff
@@ -64,7 +64,7 @@ def __init__(self, host: str = None,
 
-        use_port: str = str(self.DEFAULT_PORT)
+        use_port: str = str(self.DEFAULT_GRPC_PORT)
         if port is not None:
```
For gRPC sessions, use DEFAULT_GRPC_PORT explicitly.

andreidenissov-cog commented on neuro_san/session/grpc_concierge_session.py:
```diff
@@ -57,7 +57,7 @@ def __init__(self, host: str = None,
 
-        use_port: str = str(self.DEFAULT_PORT)
+        use_port: str = str(self.DEFAULT_GRPC_PORT)
         if port is not None:
```
For gRPC sessions, use DEFAULT_GRPC_PORT explicitly.

d1donlydfink commented on neuro_san/interfaces/concierge_session.py:
```diff
@@ -21,14 +21,17 @@ class ConciergeSession:
 
+    # Default port for the Concierge Service
+    # This port number will also be mentioned in its Dockerfile
+    DEFAULT_PORT: int = DEFAULT_HTTP_PORT
```
Seems like ConciergeSession could get these from AgentSessionConstants.
But that is out of scope.
